src/main/java/de/unijena/cheminf/mortar/model/fragmentation/FragmentationService.java

```diff
@@ -363,8 +363,9 @@ public void startPipelineFragmentation(List<MoleculeDataModel> aListOfMolecules,
                                         tmpNewFragmentFrequenciesOfMol.get(tmpChildFragment.getUniqueSmiles()) + tmpParentFragment.getFragmentFrequencyOfSpecificFragmentation(tmpPipelineFragmentationName).get(tmpChildFragment.getUniqueSmiles())
                                 );
                             } else {
-                                tmpChildFragment.getParentMolecules().clear();
-                                tmpChildFragment.getParentMolecules().add(tmpMolecule);
+                                if (tmpChildFragment.getParentMolecules().stream().noneMatch(x -> x.getUniqueSmiles().equals(tmpMolecule.getUniqueSmiles()))) {
+                                    tmpChildFragment.getParentMolecules().add(tmpMolecule);
+                                }
```

Comment:
A simple 'add' query should suffice instead of the complicated 'match' query of the set. If the object does not exist, it will be added, and if it does, the set will remain unchanged.

Reply:
I tried that but then I had the same parent molecule multiple times in the set. The question would be how the .equals() method of these objects behaves since I don't think we override it.

Reply:
Weird, I will look at this tomorrow